packages/styled-components/src/models/StyleSheetManager.js

```diff
@@ -38,7 +38,7 @@ export default function StyleSheetManager(props: Props) {
 
     if (props.sheet) {
       // eslint-disable-next-line prefer-destructuring
-      sheet = props.sheet;
+      sheet = props.sheet.rehydrate();
```

Comment:
@mu29 do you have an example of this in use? We call `rehydrateSheet` from the Sheet constructor, so realistically it should already be called by the time you pass `sheet` into `StyleSheetManager`